### PR TITLE
[WFLY-18163] Add tests for JaasSecurityRealm via a custom-realm resource

### PR DESCRIPTION
**Pull request by skyllarr**
https://issues.redhat.com/browse/WFLY-18163
Related to https://github.com/wildfly/wildfly/pull/17044

Requires:
https://issues.redhat.com/browse/WFCORE-6411

### CODE REVIEW

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * A custom {@link CallbackHandler} used in the JAAS security realm tests. It implements the
+ * {@code setSecurityInfo} method that has been historically used to populate custom handlers. Also, its {@code handle}
+ * implementation will handle any kind of credential by calling {@code toString} and then {@code toCharArray} on the opaque
+ * object.
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public class TestCallbackHandler implements CallbackHandler {
```
Comment (darranl) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java:42]: I suspect a lot of the time this is all the CallbackHandler will need to do - is there any reason we don't add a simple CallBackHandler ourselves that can be used for this scenario?

**Review by darranl — APPROVED** [review]
My comment is just for something possibly in the future, otherwise looks good.

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     */
+    public void setSecurityInfo(final Principal principal, final Object evidence) {
+        this.principal = principal;
+        this.evidence = (Evidence) evidence;
+    }
+
+    @Override
+    public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
+        if (callbacks == null)
+            throw new IllegalArgumentException("The callbacks argument cannot be null");
```
Comment (darranl) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java:64]: Minor minor issue, would prefer some parenthesis here.

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java (hunk 1 above)
Comment (Skyllarr) [reply testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java:42]: @darranl We do have a simple default callback handler implementation that gets used when none is provided by the user: https://github.com/wildfly-security/wildfly-elytron/blob/2.x/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/JaasSecurityRealm.java#L309 I added the custom callback handler here just so it gets tested as a parameter to the realm

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java (hunk 2 above)
Comment (Skyllarr) [reply testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/TestCallbackHandler.java:64]: @darranl Fixed. Thanks for review!

**Review by fjuma — APPROVED** [review]

**Comment by yersan** [comment]
Superseded by https://github.com/wildfly/wildfly/pull/17622, already merged

**Pull request closed without merging**